src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -11,81 +11,111 @@ const debug = false;
 // Brain Freeze appears to always fall after Flurry cast, but not always on same timestamp. Giving a margin here.
 const PROC_WINDOW_MS = 100;
 
+// When Glacial Spike is talented, it's OK to overwrite Brain Freeze procs if you have at least this many icicles *during* the frost bolt cast
+const ICICLE_BREAKPOINT = 3;
+
 class BrainFreeze extends Analyzer {
-	lastFlurryTimestamp;
+  lastFlurryTimestamp;
 
-	overwrittenProcs = 0;
-	expiredProcs = 0;
-	totalProcs = 0;
-	flurryWithoutProc = 0;
+  overwrittenProcs = 0;
+  okOverwrittenProcs = 0;
+  expiredProcs = 0;
+  totalProcs = 0;
+  flurryWithoutProc = 0;
 
-	on_byPlayer_applybuff(event) {
-		const spellId = event.ability.guid;
-    if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.totalProcs += 1;
-	}
+  // Tracks whether the last brain freeze generator to be cast was Ebonbolt or Frostbolt
+  wasLastGeneratorEB = false;
+
+  constructor(...args) {
+    super(...args);
+    this.glacialSpikeTalented = this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id);
+  }
 
-	on_byPlayer_refreshbuff(event) {
-		const spellId = event.ability.guid;
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.BRAIN_FREEZE.id) {
+      this.totalProcs += 1;
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
     if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.overwrittenProcs += 1;
-		this.totalProcs += 1;
-    debug && console.log("Brain Freeze proc overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      return;
+    }
+    this.totalProcs += 1;
+
+    if (!this.glacialSpikeTalented || this.wasLastGeneratorEB) {
+      this.overwrittenProcs += 1;
+      debug && console.log("Brain Freeze proc overwritten w/o GS talented or by EB @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      return;
+    }
+
+    // Get the number of icicles the player had *after* the cast of the frostbolt
+    const stacks = this.selectedCombatant.getBuff(SPELLS.ICICLES_BUFF.id).stacks || 0;
+
+    // The breakpoint is based on the number of icicles the player had *during* the cast of frostbolt, so increment by 1 to check against current icicles
+    if (stacks < ICICLE_BREAKPOINT + 1) {
```

Comment:
Just change the breakpoint to 4 instead of adding one for the sake of making the suggestion work right.

Its cleaner to just make the breakpoint correct instead of adding one here and complicating things. I would personally rather have the logic clean and easy for the next person that comes along to be able to understand it and not reference ICICLE_BREAKPOINT in the suggestion than to muddy up the logic just to be able to use ICICLE_BREAKPOINT in the suggestion.

Reply:
The reason it mentions the cast of Frostbolt is because you only cast brain freeze after a Frostbolt to shatter it. Otherwise it's a mistake to just cast the flurry and ice lance if it wasn't before the FB. So the thought process the player should have is "do I have 3 or fewer icicles as I'm casting this FB?" If so, they cast flurry right as the FB finishes, otherwise they hold the proc. Personally I believe it's easier for most people to understand "do I have 3 icicles now as I'm casting" than "will I have 4 icicles when this cast finishes". Thoughts?

Reply:
Well yes, but neither of those are relevant to the munching as both are only at five icicles (at which point if you are casting flurry without one of those two its a different mistake).

And that's the point of the comment, it explains the variable. The same person might see 4 and be used to the "during the cast" breakpoint value and think it should be 3, not knowing the order that the events happen. I did the commenting and breakpoint value for the sake of consistency, not just for the suggestion text. In my opinion (and I could be wrong), I believe the easiest and clearest way to explain it to the player (and maintainer) is by putting it in terms of what you'll be thinking while playing the spec. "Ok, I'm casting frostbolt. If I have 3 or more icicles right now, I do not cast flurry after my frost bolt" vs "Will I have 4 or more after my cast?"

Reply:
As I'm thinking, perhaps a compromise would be to move the +1 logic to another variable directly beneath the ICICLE_BREAKPOINT (something like ICICLE_AFTER_BREAKPOINT = ICICLE_BREAKPOINT +1), and then have a comment explaining the order of events and you can use either one depending on the context you're using the variable in. That way a maintainer, if the event order or breakpoint changes, can just change them at the top by seeing the comments and determining whether they need to change the breakpoint or remove the +1.

Reply:
Well hypothetically, let's say you decide to use it in more than one place, it just makes it more difficult to maintain when you use magic numbers like that. By the same token I could just completely remove icicle_breakpoint and just use 4 in the if statement and it would work fine. But by adding in the const I can keep the entire module consistent and easy to maintain (as long as the variables are explained enough)

Reply:
I dont have a problem with using the constant in the suggestion, i just dont really like that the constant doesnt represent the same value throughout the module. Cause in the suggestion it means 3 but in the if statement it means 4 (or 3+1) .... Its probably just a personal annoyance and doesnt matter in the grand scheme of things ....

Though the point i was trying to make earlier about assuming they are casting Frostbolt is that Yes I understand that they should not be casting Ebonbolt unless they have 5 Icicles and no Brain Freeze .... but people are going to do it anyway. Yes all the guides say to never cast Ebonbolt unless you have 5 icicles and need a brain freeze proc, but I can guarantee you that there will be people that choose to cast Ebonbolt on cooldown regardless. So I was wanting the suggestion to use the number 4 not 3 because I prefer to assume that not everyone is going to play correctly and i would rather the suggestions be accurate regardless of whether they are playing correctly or not.